magda-web-client/src/actions/datasetSearchActions.js

```diff
@@ -44,14 +48,17 @@ export function resetDatasetSearch(): SearchAction {
 export function fetchSearchResults(query: string): Store {
     return (dispatch: Dispatch) => {
         let url: string = config.searchApiUrl + `datasets?${query}`;
-        dispatch(requestResults(query));
```

Comment:
(comment 1/3)

You want to leave this here, _but_ you want to have it pass the `Query` object instead of the string.

So you need to change `fetchSearchResultsIfNeeded` to pass through the query object

```javascript
export function fetchSearchResultsIfNeeded(urlQueryObject: Object): Store {
     const apiQuery = parseQuery(urlQueryObject);
     return (dispatch, getState) => {
         if (shouldFetchSearchResults(getState(), urlQueryObject.q, apiQuery)) {
             return dispatch(fetchSearchResults(apiQuery, urlQueryObject));
         }
     };
 }
```

So now this function can be 
```javascript
 export function fetchSearchResults(query: string, queryObject: Object): Store {
     return (dispatch: Dispatch) => {
         let url: string = config.searchApiUrl + `datasets?${query}`;
        dispatch(requestResults(queryObject));
```